docs/howto/filtering_deep_dive.ipynb

```diff
@@ -0,0 +1,2195 @@
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of groups in entire inventory: 15\n",
+      "Single group: ptl\n",
+      "==================================================\n",
+      "Displaying information for group: ptl\n",
+      "{\n",
+      "  \"name\": \"ptl\",\n",
+      "  \"connection_options\": {},\n",
+      "  \"groups\": [],\n",
+      "  \"data\": {\n",
+      "    \"full_name\": \"Port Louis\",\n",
+      "    \"country\": \"Mauritius\",\n",
+      "    \"region\": \"amea\",\n",
+      "    \"hemisphere\": \"southern\",\n",
+      "    \"site_type\": \"primary\"\n",
+      "  },\n",
+      "  \"hostname\": null,\n",
+      "  \"port\": null,\n",
+      "  \"username\": null,\n",
+      "  \"password\": null,\n",
+      "  \"platform\": null\n",
+      "}\n",
+      "==================================================\n",
+      "Displaying data for group: ptl\n",
+      "Full Name is: Port Louis\n",
+      "Platform is: None\n",
+      "Region is: amea\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Import modules\n",
+    "from nornir import InitNornir\n",
+    "from nornir.core.filter import F\n",
+    "import json\n",
+    "\n",
+    "# Initialise nornir\n",
+    "nr = InitNornir(config_file=\"filtering_deep_dive/config.yaml\")\n",
+    "# Print the number of groups in the inventory\n",
+    "print(f\"Number of groups in entire inventory: {len(nr.inventory.groups)}\")\n",
+    "# Filter the entire group inventory on one group\n",
+    "single_group = nr.inventory.groups['ptl']\n",
+    "# Print out the group\n",
+    "print(f\"Single group: {single_group}\")\n",
+    "\n",
+    "# Dump the dictionary, assign to variable\n",
+    "single_group_data = json.dumps(single_group.dict(), indent=2)\n",
+    "# Print seperator\n",
+    "print(\"=\" * 50)\n",
+    "# Print header and host data structure\n",
+    "print(f\"Displaying information for group: {single_group}\")\n",
+    "print(f\"{single_group_data}\")\n",
+    "# Print seperator\n",
+    "print(\"=\" * 50)\n",
+    "\n",
+    "print(f\"Displaying data for group: {single_group}\")\n",
+    "# Access the full_name value, nested under the data key\n",
+    "single_group_site_code = single_group.dict()[\"data\"][\"full_name\"]\n",
+    "print(f\"Full Name is: {single_group_site_code}\")\n",
+    "# Access the platform value\n",
+    "single_group_platform = single_group.dict()[\"platform\"]\n",
+    "print(f\"Platform is: {single_group_platform}\")\n",
+    "# Access the os_version value, nested under the data key\n",
+    "single_group_region = single_group.dict()[\"data\"][\"region\"]\n",
+    "print(f\"Region is: {single_group_region}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "creative-hearts",
+   "metadata": {},
+   "source": [
+    "By now, you should have an understanding of the following regarding inventory custom data:  \n",
+    "    - __What is nornir custom data?__  \n",
```

Comment:
check how this is being formatted, not sure if it was your intention to render the underscores or if you were looking for italic or something else

Reply:
Thanks for picking this up. Yep, I meant for this to be italic and have now resolved it.